cni/README.md

```diff
@@ -1,96 +1,55 @@
-# Istio CNI plugin
+# Istio CNI Node Agent
 
-For application pods in the Istio service mesh, all traffic to/from the pods needs to go through the
-sidecar proxies (istio-proxy containers).  This `istio-cni` Container Network Interface (CNI) plugin will
-set up the pods' networking to fulfill this requirement in place of the current Istio injected pod `initContainers`
-`istio-init` approach.
+The Istio CNI Node Agent is responsible for several things
 
-This is currently accomplished via configuring the iptables rules in the netns for the pods.
+- Install an Istio CNI plugin on each node's filesystem, updating that node's CNI config in e.g (`/etc/cni/net.d`), and watching the config and binary paths to reinstall if things are modified.
+- In sidecar mode, CNI plugin can configure sidecar networking for pods when they are scheduled by the container runtime, using iptables. The CNI handling the netns setup replaces the current Istio approach using a `NET_ADMIN` privileged `initContainers` container, `istio-init`, injected in the pods along with `istio-proxy` sidecars. This removes the need for a privileged, `NET_ADMIN` container in the Istio users' application pods.
+- In ambient mode, CNI plugin does not configure any networking, but will push events to an ambient watch server running in the agent Daemonset on new pod. The ambient server will find the pod netns and configure networking inside that pod via iptables. The ambient server will additionally watch enabled namespaces, and enroll already-started-but-newly-enrolled pods in a similar fashion.
```

Comment:
```suggestion
- In ambient mode, CNI plugin does not configure any networking, but will synchronously push events to an ambient watch server running in the agent Daemonset on new pod. The ambient server will find the pod netns and configure networking inside that pod via iptables. The ambient server will additionally watch enabled namespaces, and enroll already-started-but-newly-enrolled pods in a similar fashion.
```

nit: per both design and implementation, this event handler should be synchronous/blocking one, right?

Reply:
Yes, this is a bad wording.